scanner/src/main/java/ca/on/oicr/gsi/runscanner/scanner/processor/DefaultPromethion.java

```diff
@@ -0,0 +1,81 @@
+package ca.on.oicr.gsi.runscanner.scanner.processor;
+
+import ca.on.oicr.gsi.runscanner.dto.NanoporeNotificationDto;
+import ca.on.oicr.gsi.runscanner.dto.NotificationDto;
+import ca.on.oicr.gsi.runscanner.dto.type.HealthType;
+import ch.systemsx.cisd.hdf5.HDF5FactoryProvider;
+import ch.systemsx.cisd.hdf5.IHDF5StringReader;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.io.File;
+import java.io.IOException;
+import java.time.ZonedDateTime;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.TimeZone;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class DefaultPromethion extends RunProcessor {
```

Comment:
Is this going to be PromethION-specific? Does there need to be a separate processor for MinION?

Reply:
As far as I can tell, the PromethION and MinION fast5 files use the same fields to store different information, and save them in different directory structures on disk. They will need separate RunProcessors.

Reply:
It probably makes sense to have a `BaseNanoporeProcessor` that includes the method that handles the Fast5 and then have a `PromethionProcessor` that extends it. The `DefaultX` thing was because there was only one processor for Illumina and PacBio, but they may have had other instruments. Particularly, the Illumina GenomeAnalyzers would need a totally independent processor, but we decided not to bother writing one for that data.

Reply:
> the PromethION and MinION fast5 files use the same fields to store different information, and save them in different directory structures on disk

:man_facepalming: